City-of-Helsinki/drupal-hdbt
# UHF-2876: Header style changes

# Header style changes [UHF-2876](https://helsinkisolutionoffice.atlassian.net/browse/UHF-2876)
 "Minor header style changes"

## What was done
* Requested style changes (and extrapolated minor changes based on those)
  * Main idea: menu navigation links are underlined unless hovered of focused
  * Active links are bolded
    * Some active links also have thicker border in main nav
* Cleaned header bottom menu SCSS code (less nesting)
* Added more class names to menu twig to avoid indirect element selectors
* Fixed some accessibility issues from all menus 
  * aria-expanded -bug
* Improved language switcher style and functionality
  * If additional translations are available, they're shown in menu (russian)
  * Accessibility fixes based on accessibility meeting notes on 16.2.2022
    * lang -attribute always available
    * current translation works as a link with `aria-current="page"`
    * disabled translations contain an `title`-attribute and cursor style change (`not-allowed` -cursor)
* Switched code to use aria-attributes where they are available to keep accessibility and visuality in sync
* Sprinkled a bit more animations on arrows and chevrons
* Moved all arrow and chevron animations to be conditional to `prefers-reduced-motion` check
* Moved multiple navigation related codes from two files single file for clarity (same class was referenced in two places)
* Disabled external link icon from navigation menus until we have the whitelist feature working and actual external links marked

## How to test
* Get branch
* Check main, top, branding, side, dropdown and footer navigations.
* Test also system settings > accessibility > display > reduce motion for prefers-reduced-motion checking
* Check different language features in translation menu, russian, disabled languages, etc
